### PR TITLE
Remove details about chalk and point to Chalk Book instead.

### PR DESCRIPTION
**Pull request by crlf0710**
Now that these contents have been moved to the [Chalk Book](https://rust-lang.github.io/chalk/book), this PR remove these contents and point to that instead.

### CODE REVIEW

**Comment by tshepang** [comment]
wow, wasn't aware there's so much chalk doc here

File: src/traits/lowering-module.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # The lowering module in rustc
```
Comment (mark-i-m) [inline src/traits/lowering-module.md:1]: I think this was all removed from rustc

File: src/SUMMARY.md [hunk 2]
```diff
@@ -72,24 +72,13 @@
...
-            - [The lowering module in rustc](./traits/lowering-module.md)
-            - [Lowering rules](./traits/lowering-rules.md)
-            - [Well-formedness checking](./traits/wf.md)
-        - [Canonical queries](./traits/canonical-queries.md)
-            - [Canonicalization](./traits/canonicalization.md)
-        - [The SLG solver](./traits/slg.md)
-        - [An Overview of Chalk](./traits/chalk-overview.md)
-        - [Bibliography](./traits/bibliography.md)
+        - [Chalk-based trait solving](./traits/chalk.md)
+        - [Region constraints](./traits/regions.md)
```
Comment (mark-i-m) [inline src/SUMMARY.md:80]: I would just remove this subchapter and delete `./traits/region.md`

File: src/traits/associated-types.md [hunk 3]
```diff
@@ -1,168 +0,0 @@
-# Equality and associated types
```
[the remark this answers is not in the document]
Comment (detrumi) [reply src/traits/associated-types.md:1]: I think this is mostly on chalk's side, so just linking to the chalk book for this seems fine.

File: src/appendix/code-index.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ Item            |  Kind    | Short description           | Chapter            |
...
 `rustc_ast::token_stream::TokenStream` | struct | An abstract sequence of tokens, organized into `TokenTree`s | [The parser], [Macro expansion] | [src/librustc_ast/tokenstream.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/tokenstream/struct.TokenStream.html)
 `TraitDef` | struct | This struct contains a trait's definition with type information | [The `ty` modules] |  [src/librustc_middle/ty/trait_def.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/trait_def/struct.TraitDef.html)
-`TraitRef` | struct | The combination of a trait and its input types (e.g. `P0: Trait<P1...Pn>`) | [Trait Solving: Goals and Clauses], [Trait Solving: Lowering impls]  |  [src/librustc_middle/ty/sty.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TraitRef.html)
+`TraitRef` | struct | The combination of a trait and its input types (e.g. `P0: Trait<P1...Pn>`) | [Chalk Book: Goals and Clauses], [Chalk Book: Lowering impls]  |  [src/librustc_middle/ty/sty.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TraitRef.html)
```
[the remark this answers is not in the document]
Comment (detrumi) [reply src/appendix/code-index.md:30]: Right. Let's keep the goals and clauses section in both books for now, most of the other subsections can be moved over.

File: src/traits/lowering-module.md (hunk 1 above)
Comment (detrumi) [reply src/traits/lowering-module.md:1]: There's a WIP rustc PR that's going to add a new lowering module, we should add/fill this section again when that lands.

**Comment by detrumi** [comment]
> I don't know if there is any non-chalk-specific knowledge in these chapters that we should condense and keep in some form

Let's see, going over the sections:
- Lowering to logic: This has a nice short introduction which won't change much (since it's based on a blog post, and talks about the general ideas), so the introduction can stay. And the "goals and clauses" subsection seems relevant to rustc too
- Canonical queries: this talks about trait queries in rustc, so this should stay here (and possibly not even add it to the chalk book)
- SLG solver: this talks about chalk internals, so that can go to the chalk book
- chalk overview: this can just be replaced by a link to the chalk book

> I don't know if any of the chalk-rustc integration content is still relevant. I understood it to be all out of date. So perhaps we should just mention that there is some experimental effort to do integration and leave it at that?

The WF section states "The new-style WF checking has not been implemented in rustc yet", so I think there isn't much. Mentioning something like that sounds like a good plan.

**Comment by crlf0710** [comment]
Addressed the comments above. Please review again to see if there's more to be adjusted.

**Comment by crlf0710** [comment]
Wait, https://rust-lang.github.io/chalk/book/ is now 404. Did we have a url change or something?

**Comment by crlf0710** [comment]
Comments are addressed, and seems CI passes now.

**Pull request merged**